generic_record.go

```diff
@@ -48,3 +50,31 @@ func (this *GenericRecord) Set(name string, value interface{}) {
 func (this *GenericRecord) Schema() Schema {
 	return this.schema
 }
+
+// Returns a JSON representation of this GenericRecord.
+func (this *GenericRecord) String() (string, error) {
```

Comment:
[Stringer](https://golang.org/pkg/fmt/#Stringer) is defined as `String() string` so I suppose it should be this way. I don't see any reason json marshalling would ever return a non-nil error here so I think we could just replace this with `if err != nil { panic(err) }`